i2c/mcp9808_i2c/mcp9808_i2c.c

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright (c) 2020 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include "pico/stdlib.h"
+#include "pico/binary_info.h"
+#include "hardware/i2c.h"
+
+/* Example code to talk to a MCP9808 ±0.5°C Digital temperature Sensor
+   
+   This reads and writes to registers on the board. 
+
+   Connections on Raspberry Pi Pico board, other boards may vary.
+
+   GPIO PICO_DEFAULT_I2C_SDA_PIN (On Pico this is GP4 (physical pin 6)) -> SDA on MCP9808 board
+   GPIO PICO_DEFAULT_I2C_SCK_PIN (On Pico this is GP5 (physcial pin 7)) -> SCL on MCP9808 board
+   Vsys (physical pin 39) -> VDD on MCP9808 board
+   GND (physical pin 38)  -> GND on MCP9808 board
+
+*/
+//The bus address is determined by the state of pins A0, A1 and A2 on the MCP9808 board
+static uint8_t ADDRESS = 0x18;
+
+const uint8_t WRITE_MODE = 0xFE;
+const uint8_t READ_MODE = 0xFF;
+
+
+//hardware registers
+
+const uint8_t REG_POINTER = 0x00;
+const uint8_t REG_CONFIG = 0x01;
+const uint8_t REG_TEMP_UPPER = 0x02;
+const uint8_t REG_TEMP_LOWER = 0x03;
+const uint8_t REG_TEMP_CRIT = 0x04;
+const uint8_t REG_TEMP_AMB = 0x05;
+const uint8_t REG_MAN_ID = 0x06;
+const uint8_t REG_DEV_ID = 0x07;

```

Comment:
Is it worth including these unused values? :shrug:

Reply:
Have cut those out now :)

Reply:
What about the config and resolution register? I haven't used them in my code but should I leave them there in case someone wants to use them or are we going for complete minimalism?

Reply:
Question for @aallan ...

Reply:
Me? Gods no. I'm not making code decisions like that, that's a @kilograham question.

Reply:
If you do include them then perhaps add some commented out example code showing how they *could* be used?

Reply:
> If you do include them then perhaps add some commented out example code showing how they could be used?

That I will say "no" to, no commented out code in the examples.